flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducerTest.java

```diff
@@ -325,16 +325,20 @@ public void go() throws Exception {
 		};
 		moreElementsThread.start();
 
-		moreElementsThread.trySync(deadline.timeLeftIfAny().toMillis());
 		assertTrue("Producer should still block, but doesn't", moreElementsThread.isAlive());
 
 		// consume msg-2 from the queue, leaving msg-3 in the queue and msg-4 blocked
+		while (producer.getPendingRecordFutures().size() < 2) {
```

Comment:
How about to move it to the line where the previous `moreElementsThread.trySync` was, so we can retain the behavior to check the `moreElementsThread.isAlive()` after sleeping for a period?

Reply:
I think that only makes sense when looking at the diff and treating this change as a drop-in replacement.

What actually happened was that I removed the syncs because they are pointless; they don't provide any guarantees that anything of interest has happened. In fact the same could be said the for isAlive() checks; after all the thread can die just after the check. But for the rare case where they do find an issue they provide a more interesting error message than a TimeoutException.

After the removal the test started failing since it relies on elements being processed in the background. Hence the loops were added, immediately before the code that relies on some processing to have happened (`getPendingRecordFutures().get(2)`).

Putting the loop before the isAlive() calls renders these calls basically pointless since it doesn't make sense for the thread to die _after having finished what it's supposed to do_, which the loop verifies.

Reply:
Thanks for the explanation. Especially agreed that "In fact the same could be said the for isAlive() checks". This, however, means that the first 2 checks for `isAlive()` would not fail in cases it is supposed to.
Imagine that there is a bug that msg-3 does not block the thread before it is consumed from the producer, this case is still possible to pass because the checks may happen before the thread dies.
It is not a problem introduced by thins PR though and can be fixed in another task.